src/audit/forbidden_uses.rs

```diff
@@ -0,0 +1,123 @@
+use github_actions_models::common::{RepositoryUses, Uses};
+
+use super::{audit_meta, Audit, AuditState, Finding, Step};
+use crate::config::Config;
+use crate::finding::{Confidence, Persona, Severity};
+use crate::models::{uses::RepositoryUsesExt, CompositeStep};
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+struct ForbiddenUsesConfig {
+    allow: Vec<String>,
+}
+
+impl Default for ForbiddenUsesConfig {
+    fn default() -> Self {
+        Self {
+            allow: vec![
+                String::from("actions/checkout"),
+                String::from("actions/cache"),
+                String::from("actions/upload-artifact"),
+            ],
+        }
+    }
+}
+
+impl ForbiddenUsesConfig {
+    fn is_uses_allowed(&self, uses: &RepositoryUses) -> bool {
+        self.allow
+            .iter()
+            .any(|allowlist_entry| uses.matches(allowlist_entry))
+    }
+
+    #[inline]
+    fn is_uses_denied(&self, uses: &RepositoryUses) -> bool {
+        !self.is_uses_allowed(uses)
+    }
+}
+
+pub(crate) struct ForbiddenUses {
+    config: ForbiddenUsesConfig,
+}
+
+audit_meta!(ForbiddenUses, "forbidden-uses", "fobidden action used");
+
+impl ForbiddenUses {
+    pub fn evaluate_allowlist<'u>(&self, uses: &Uses) -> Option<(&'u str, Severity, Persona)> {
+        let Uses::Repository(repo_uses) = uses else {
+            return None;
+        };
+
+        self.config.is_uses_denied(repo_uses).then_some((
+            "action is not on the allowlist",
+            Severity::Medium,
+            Persona::default(),
```

Comment:
I think this needs to be the "auditor" persona when the user doesn't supply a configuration, since otherwise this will be _way_ too noisy for typical users.